src/gdb/debugexception.S

```diff
@@ -0,0 +1,207 @@
+/*
```

Comment:
Since this is xtensa assembly is should go under src/arc/xtensa/gdb/ and we can keep generic gdb code in src/gdb.

Reply:
@lgirdwood, I can move it there if you want, no problem. However I would personally keep all GDB related source code in one "GDB" folder. This way we have better consistency (all sources in one folder), less work in maintenance (fewer config/make files) plus, I don't really see any reason to keep assembly code separately. Moreover in following days I am going to commit some C code with mixed in "inline assembly".

I think you proposal is OK in case where we would use some library or Xtensa package. In this case I vote to keep it simply in src/gdb/*  

Does it make sense for you? :)

Reply:
I think it's best moving under arch/xtensa/gdb. This way we keep all xtensa code in the one place so  it can easily be ignored by people working on non xtensa architectures. 
However, we still do need the top level src/gdb for generic C parts (which should be architecture agnostic) and we can also put any xtensa specific C code in arch/xtensa/gdb too.

Overall aim is to provide GDB support but in a architecture agnostsic manner. The kernel parts will be arch agnostic as well.

Reply:
@mrajwa 
Probably also src/include/xtensa-defs.h should be moved to xtensa arch headers.

Reply:
@jajanusz, yes I think so...as well as init.S.